tests/new_upgrades/test_activation_key.py

```diff
@@ -0,0 +1,129 @@
+    with SharedResource(target_sat.hostname, upgrade_action, target_sat=target_sat) as sat_upgrade:
+        test_data = Box(
+            {
+                'target_sat': target_sat,
+                'test_name': None,
+                'org': None,
+                'product': None,
+                'repo': None,
+                'cv': None,
+                'ak': None,
+            }
+        )
+        test_name = f'ak_upgrade_{gen_alpha()}'
+        test_data.test_name = test_name
+        org = target_sat.api.Organization(name=f"{test_name}_org").create()
+        test_data.org = org
+        product = target_sat.api.Product(organization=org, name=f'{test_name}_prod').create()
+        test_data.product = product
+        custom_repo = target_sat.api.Repository(
+            product = product,
+            name = f'{test_name}_repo',
+            url=settings.repos.yum_1.url,
+            content_type = 'yum',
+        ).create()
+        test_data.repo = custom_repo
+        custom_repo.sync()
+        cv = target_sat.publish_content_view(org, [custom_repo], f'{test_name}_cv')
+        assert len(cv.read_json()['versions']) == 1
+        test_data.cv = cv
+        ak = target_sat.api.ActivationKey(
+            content_view=cv, organization=org, name=f'{test_name}_ak'
+        ).create()
+        test_data.ak = ak
+        ak.host_collection.append(target_sat.api.HostCollection(organization=org, name=f'{test_name}_hc').create())
+        ak = ak.update(['host_collection'])
+        assert len(ak.host_collection) == 1
+        sat_upgrade.ready()
+        target_sat._session = None
+        yield test_data
+    
+
+@pytest.mark.content_upgrades
+def test_ak_upgrade_scenario(ak_upgrade_setup):
+    """After Upgrade, Activation keys entities remain the same and
+    all their functionality works.
+
+    :id: postupgrade-a7443b54-eb2e-497b-8a50-92abeae01496
+
+    :steps:
+        1. Postupgrade, Verify activation key has same entities associated.
+        2. Update existing activation key with new entities.
+        3. Delete activation key.
+
+    :expectedresults: Activation key's entities should be same after upgrade and activation
+        key update and delete should work.
+
+    :BlockedBy: SAT-28048, SAT-28990
+    """
+    target_sat = ak_upgrade_setup.target_sat
+    # pre_test_name = dependent_scenario_name
+    org = target_sat.api.Organization().search(query={'search': f'name={ak_upgrade_setup.org.name}'})[0]
+    ak = target_sat.api.ActivationKey(organization=org.id).search(
+        query={'search': f'name={ak_upgrade_setup.ak.name}'}
+    )[0]
+    cv = target_sat.api.ContentView(organization=org.id).search(
+        query={'search': f'name={ak_upgrade_setup.cv.name}'}
+    )[0]
+    custom_repo = target_sat.api.Repository(organization=org.id).search(query={'search': f'name={ak_upgrade_setup.repo.name}'})[0]
+    assert f'{ak_upgrade_setup.test_name}_ak' == ak.name
+    assert f'{ak_upgrade_setup.test_name}_cv' == cv.name
```

Comment:
It would be nice to assert the CV (and possibly the LCE too) remained associated with the AK.
```suggestion
    assert f'{ak_upgrade_setup.test_name}_ak' == ak.name
    assert f'{ak_upgrade_setup.test_name}_cv' == cv.name
    assert ak.content_view == cv  # or ak.content_view.id == cv.id
```